alicloud/resource_alicloud_drds_instance.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAliCloudDRDSInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliCloudDRDSInstanceCreate,
+		Read:   resourceAliCloudDRDSInstanceRead,
+		Update: resourceAliCloudDRDSInstanceUpdate,
+		Delete: resourceAliCloudDRDSInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"description": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateStringLengthInRange(1, 129),
+			},
+			"type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAllowedStringValue([]string{string(PrivateType), string(PrivateType_)}),
+			},
+			"zone_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"specification": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"pay_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAllowedStringValue([]string{string(DRDSInstancePostPayType), string(DRDSInstancePrePayType)}),
+			},
+			"vswitch_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"instance_series": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAliCloudDRDSInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	drdsService := DrdsService{client}
+
+	req := drds.CreateCreateDrdsInstanceRequest()
+	req.Description = d.Get("description").(string)
+	req.Type = d.Get("type").(string)
+	req.ZoneId = d.Get("zone_id").(string)
+	req.Specification = d.Get("specification").(string)
+	req.PayType = d.Get("pay_type").(string)
+	req.VswitchId = d.Get("vswitch_id").(string)
+	req.InstanceSeries = d.Get("instance_series").(string)
+	req.Quantity = "1"
+	response, err := drdsService.CreateDrdsInstance(req)
+	idList := response.Data.DrdsInstanceIdList.DrdsInstanceId
+	if err != nil || len(idList) != 1 {
+		return fmt.Errorf("failed to create DRDS instance with error: %s", err)
+	}
+	d.SetId(idList[0])
+	return resourceAliCloudDRDSInstanceUpdate(d, meta)
+
+}
+
+func resourceAliCloudDRDSInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	drdsService := DrdsService{client}
+	update := false
+	req := drds.CreateModifyDrdsInstanceDescriptionRequest()
+	req.DrdsInstanceId = d.Id()
+	if d.HasChange("description") && !d.IsNewResource() {
+		update = true
+		req.Description = d.Get("description").(string)
+	}
+	if update {
+		_, err := drdsService.ModifyDrdsInstanceDescription(req)
+		if err != nil {
+			return fmt.Errorf("failed to update Drds instance with error: %s", err)
+		}
+	}
```

Comment:
I think line 82-86 and 91-96 can be moved into the line 87 block.

Reply:
No need to modify here